com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/dependency.py

```diff
@@ -23,22 +21,21 @@ def __init__(self, topology_packages):
         dir = os.path.dirname(os.path.abspath(my_module.__file__))
         dir = os.path.dirname(dir)
         self._streamsx_topology_dir = dir
-        self._topology_packages = topology_packages
         
-    def add_dependencies(self, module):
+    def add_dependencies(self, module, included_packages=set(), excluded_packages=set()):
```

Comment:
Shouldn't this be:
```Python
def add_dependencies(self, module, included_packages=None, excluded_packages=None):
    if included_packages is None:
        included_packages = set()
    if excluded_packages is None:
        excluded_packages = set()
```

Otherwise, `included_packages` and `excluded_packages` becomes state which gets preserved over calls to add_dependencies.